From 8f3cd5ef835c52e645b83a906b4d9d0fa265a814 Mon Sep 17 00:00:00 2001
From: Jianqun Xu <jay.xu@rock-chips.com>
Date: Wed, 19 Dec 2018 14:09:35 +0800
Subject: [PATCH] arm64: dts: rockchip: add board dt file for rk1808 x4 evb

RK1808 x4 evb is a board with 4 rk1808 running on independent.
And it's designed with 32MB sfc norflash.

Change-Id: I809e9cf1c217d84e4dd9b79213db317751838ade
Signed-off-by: Jianqun Xu <jay.xu@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/Makefile         |  1 +
 .../arm64/boot/dts/rockchip/rk1808-evb-x4.dts | 54 +++++++++++++++++++
 2 files changed, 55 insertions(+)
 create mode 100644 arch/arm64/boot/dts/rockchip/rk1808-evb-x4.dts

diff --git a/arch/arm64/boot/dts/rockchip/Makefile b/arch/arm64/boot/dts/rockchip/Makefile
index 7ee5ab0b6a92..b47704310993 100644
--- a/arch/arm64/boot/dts/rockchip/Makefile
+++ b/arch/arm64/boot/dts/rockchip/Makefile
@@ -10,6 +10,7 @@ dtb-$(CONFIG_ARCH_ROCKCHIP) += px30-evb-ddr4-v10.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += px30-evb-ext-rk618.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += px30-z7-a0-rk618-dsi.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk1808-evb-v10.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk1808-evb-x4.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk1808-fpga.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3308-ai-va-v10.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3308-evb-amic-v10.dtb
diff --git a/arch/arm64/boot/dts/rockchip/rk1808-evb-x4.dts b/arch/arm64/boot/dts/rockchip/rk1808-evb-x4.dts
new file mode 100644
index 000000000000..ef657a361f5e
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/rk1808-evb-x4.dts
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2018 Fuzhou Rockchip Electronics Co., Ltd
+ */
+
+/dts-v1/;
+#include <dt-bindings/display/drm_mipi_dsi.h>
+#include "rk1808-evb.dtsi"
+
+/ {
+	model = "Rockchip RK1808 EVB X4 Board";
+	compatible = "rockchip,rk1808-evb-x4", "rockchip,rk1808";
+
+	chosen {
+		bootargs = "earlycon=uart8250,mmio32,0xff550000 console=ttyFIQ0 root=PARTUUID=614e0000-0000 rootwait swiotlb=1 kpti=0";
+	};
+};
+
+&emmc {
+	status = "disabled";
+};
+
+&npu {
+	status = "disabled";
+};
+
+&rk809_codec {
+	status = "disabled";
+};
+
+&route_dsi {
+	status = "disabled";
+};
+
+&sdmmc {
+	status = "disabled";
+};
+
+&sdio {
+	status = "disabled";
+};
+
+&sfc {
+	status = "okay";
+};
+
+&tsadc {
+	rockchip,hw-tshut-mode = <1>; /* tshut mode 0:CRU 1:GPIO */
+	rockchip,hw-tshut-polarity = <0>; /* tshut polarity 0:LOW 1:HIGH */
+	pinctrl-names = "init", "default";
+	pinctrl-0 = <&tsadc_otp_gpio>;
+	pinctrl-1 = <&tsadc_otp_out>;
+	status = "okay";
+};
-- 
2.35.3

